pkg/controller/oneagent/util.go

```diff
@@ -127,7 +127,7 @@ func getToken(secret *corev1.Secret, key string) (string, error) {
 		return "", err
 	}
 
-	return string(value), nil
+	return strings.TrimSpace(value), nil
```

Comment:
One could use https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L4992
`Secret.stringData` to have the `string` representation.

Reply:
`StringData` is write-only and only used for creating secrets. Don't see viable alternatives to `strings.TrimSpace` or `bytes.TrimSpace`.

Reply:
Wasn't aware this fact.

Reply:
My IDE failed to give this error on compilation. Fixed it, should be able to catch it next time.